v1/server/server.go

```diff
@@ -2677,31 +2726,44 @@ func (s *Server) updateNDCache(enabled bool) {
 	s.ndbCacheEnabled = enabled
 }
 
-func stringPathToDataRef(s string) (r ast.Ref) {
+func stringPathToDataRef(s string) (ast.Ref, error) {
 	result := ast.Ref{ast.DefaultRootDocument}
-	return append(result, stringPathToRef(s)...)
+	r, err := stringPathToRef(s)
+	if err != nil {
+		return nil, err
+	}
+	return append(result, r...), nil
 }
 
-func stringPathToRef(s string) (r ast.Ref) {
+func stringPathToRef(s string) (ast.Ref, error) {
+	r := ast.Ref{}
```

Comment:
[nit] Isn't an ast.Ref{} a slice? I'd expect we can return nil here. But this is OK.